src/ServiceProvider.php

```diff
@@ -21,6 +25,27 @@ public function boot()
         Queue::looping(function () {
             app('log')->reset();
         });
+
+        /**
+         * Add the trace ID to context if its not already present.
+         *
+         * Web requests will have this value set by the load balancer and so will add it to context.
+         * Queue jobs will have any context persisted when they are dispatched, so for jobs that are
+         * dispatched from a web request (ie not a cron) will already have this value hydrated and
+         * won't try to set another value. This will also work across fan-out jobs.
+         * Crons won't have this available ever.
+         *
+         * Any values added to context will also end up in our logs under the `extra` key. This is
+         * useful for following code across services for a single web request.
+         */
+        if (array_key_exists('HTTP_X_AMZN_TRACE_ID', $_SERVER)) {
+            Context::addIf('X-Amzn-Trace-Id', $_SERVER['HTTP_X_AMZN_TRACE_ID']);
+        }
+
+        // For any outgoing request, carry along the AWS trace ID for better observability across services
+        Http::globalRequestMiddleware(
+            fn (RequestInterface $request) => $request->withHeader('X-Amzn-Trace-Id', Context::get('X-Amzn-Trace-Id'))
+        );
```

Comment:
Block: no tests to cover this change. Also looks like it would probably add empty header values if the value is not in context?

Reply:
I could add a test, but its a bit redundant I think? It's just testing the framework does what the methods say they do

Reply:
Oh yeah, I'm more saying adding a test for this is not far from adding a test that something like `Model::query()->where('the_answer', 42)` results in sql like `select * from models where the_answer = 42`. Sure its a new line that we wrote, but its not very complex and is using straight up framework features.

Do you still feel strongly?